clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/ttl/VeniceRmdTTLFilter.java

```diff
@@ -125,7 +152,11 @@ boolean filterByTTLandMaybeUpdateValue(final INPUT_VALUE value) {
         valueSerializerCache.computeIfAbsent(valueSchemaId, this::generateValueSerializer);
     RecordSerializer<GenericRecord> rmdSerializer =
         rmdSerializerCache.computeIfAbsent(rmdVersionId, this::generateRmdSerializer);
-    updateValuePayload(value, valueSerializer.serialize(valueRecord));
+    try {
+      updateValuePayload(value, sourceVersionCompressor.compress(valueSerializer.serialize(valueRecord)));
```

Comment:
```
props.getBoolean(KAFKA_INPUT_COMPRESSION_BUILD_NEW_DICT_ENABLED, true);
```
Repush will always build a new dict for the repushed version, so shall we read the dictionary in the new version instead for the compression?

Reply:
Hmm, would that be available in the mapper / reducer phase? As far as I know this filter is invoked in both mapper and reducer today. 

Also, what'd be the difference for reading for the new dictionary? I am a bit new to ZSTD_WITH_DICT so I don't know that difference.
Also I am wondering, if dest version changed compression strategy, would it still work?

Reply:
Using newer dictionary would give better compression, but not a must. If dest changed compression strategy, we should fail the repush?

Reply:
I think we can build a new compressor for the dest version as the regular push can get a proper compressor in the reducer phase.
You can check the related code.